javascript/javascript1/week4/assistantvoice.js

```diff
@@ -0,0 +1,49 @@
+let name = 'a mentioned name';
+let givingName = '';
+let todo = [];
+
+function getReply(order) {
+	//let name;
+	if (order.includes('hello my name is')) {
+		name = order.replace('hello my name is', '');
+		return `Nice to meet you ${name}`;
+	}
+	if (order.includes('what is your name?') && name != givingName) {
+		let name;
+		let givingName;
+		console.log('Who are you?');
+	} else if (order.includes('what is your name?') && name === givingName) {
+		console.log('I am ' + givingName);
+	}
```

Comment:
This is weird. You already have the name stored at `name`. Just check if the order includes `what is your name` then return the name.
```js
  if (command === 'what is my name?') {
    return `Your name is ${name}`;
  }
```